pkg/ccl/changefeedccl/enriched_source_provider.go

```diff
@@ -192,6 +193,51 @@ func (p *enrichedSourceProvider) KafkaConnectJSONSchema() kcjsonschema.Schema {
 	return kafkaConnectJSONSchema
 }
 
+func (p *enrichedSourceProvider) GetProtobuf(
+	evCtx eventContext, updated, prev cdcevent.Row,
+) (*changefeedpb.EnrichedSource, changefeedpb.Op, error) {
+	var op changefeedpb.Op
+	md := updated.Metadata
+	tableInfo, ok := p.sourceData.tableSchemaInfo[md.TableID]
+	if !ok {
+		return nil, op, errors.AssertionFailedf("table %d not found in tableSchemaInfo", md.TableID)
+	}
+
+	src := &changefeedpb.EnrichedSource{
+		JobId:              p.sourceData.jobID,
+		ChangefeedSink:     p.sourceData.sink,
+		DbVersion:          p.sourceData.dbVersion,
+		ClusterName:        p.sourceData.clusterName,
+		ClusterId:          p.sourceData.clusterID,
+		SourceNodeLocality: p.sourceData.sourceNodeLocality,
+		NodeName:           p.sourceData.nodeName,
+		NodeId:             p.sourceData.nodeID,
+		Origin:             originCockroachDB,
+		DatabaseName:       tableInfo.dbName,
+		SchemaName:         tableInfo.schemaName,
+		TableName:          tableInfo.tableName,
+		PrimaryKeys:        tableInfo.primaryKeys,
+	}
+
+	if p.opts.mvccTimestamp {
+		src.MvccTimestamp = evCtx.mvcc.AsOfSystemTime()
+	}
+	if p.opts.updated {
+		src.TsNs = evCtx.updated.WallTime
+		src.TsHlc = evCtx.updated.AsOfSystemTime()
+	}
+
+	if updated.IsDeleted() {
+		op = changefeedpb.Op_OP_DELETE
+	} else if prev.IsDeleted() || !prev.IsInitialized() {
+		op = changefeedpb.Op_OP_CREATE
+	} else {
+		op = changefeedpb.Op_OP_UPDATE
+	}
```

Comment:
can you reuse this stuff and then map those constants to the pb ones 
https://github.com/massimo-ua/cockroach/blob/1148c1db7292e7824b5aa6c93ff385b5f7ccbebb/pkg/ccl/changefeedccl/encoder_json.go#L562-L582